tests/e2e/vsphere_volume_expansion.go

```diff
@@ -199,8 +215,406 @@ var _ = ginkgo.Describe("Volume Expansion Test", func() {
+		currentPvcSize := pvclaim.Spec.Resources.Requests[v1.ResourceStorage]
+		newSize := currentPvcSize.DeepCopy()
+		newSize.Sub(resource.MustParse("100Mi"))
+		framework.Logf("currentPvcSize %v, newSize %v", currentPvcSize, newSize)
+		_, err := expandPVCSize(pvclaim, newSize, client)
+		gomega.Expect(err).To(gomega.HaveOccurred())
+	})
+
+	/*
+			Verify online volume expansion multiple times on the same PVC
+
+			1. Create StorageClass with allowVolumeExpansion set to true.
+			2. Create PVC which uses the StorageClass created in step 1.
+			3. Wait for PV to be provisioned.
+		    4. Wait for PVC's status to become Bound and note the PVC size
+			5. Create POD using the above created PVC
+			6. In a loop of 10, modify PVC's size by adding 1 Gb at a time to trigger online volume expansion.
+			7. Wait for file system resize to complete.
+			8. Verify the PVC Size should increased by 10Gi
+			9. Make sure file system has increased
+	*/
+	ginkgo.It("[csi-block-vanilla] Verify online volume expansion multiple times on the same PVC", func() {
+
+		ginkgo.By("Invoking Test for Volume Expansion")
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		volHandle, pvclaim, pv, storageclass := createSCwithVolumeExpansionTrueAndDynamicPVC(f, client, namespace)
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, storageclass.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = e2eVSphere.waitForCNSVolumeToBeDeleted(pv.Spec.CSI.VolumeHandle)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		pod := createPODandVerifyVolumeMount(f, client, namespace, pvclaim, volHandle)
+		originalFSSize, err := getFSSizeMb(f, pod)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			// Delete POD
+			ginkgo.By(fmt.Sprintf("Deleting the pod %s in namespace %s", pod.Name, namespace))
+			err := fpod.DeletePodWithWait(client, pod)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			ginkgo.By("Verify volume is detached from the node")
+			isDiskDetached, err := e2eVSphere.waitForVolumeDetachedFromNode(client, pv.Spec.CSI.VolumeHandle, pod.Spec.NodeName)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(isDiskDetached).To(gomega.BeTrue(), fmt.Sprintf("Volume %q is not detached from the node %q", pv.Spec.CSI.VolumeHandle, pod.Spec.NodeName))
+		}()
+
+		increaseVolumeMultipleTimes(ctx, f, client, namespace, volHandle, originalFSSize, pvclaim, pod)
+	})
+
 })
 
+//increaseVolumeMultipleTimes this method increases the same volume multiple times and verifies PVC and Filesystem size
+func increaseVolumeMultipleTimes(ctx context.Context, f *framework.Framework, client clientset.Interface, namespace string, volHandle string, originalSizeInMb int64, pvclaim *v1.PersistentVolumeClaim, pod *v1.Pod) {
```

Comment:
mention this is for online resize only

Reply:
Updated as "increaseOnlineVolumeMultipleTimes"